File: java/src/main/java/parser/ast/Block.java
package parser.ast;

import parser.util.ParseException;
import parser.util.PeekTokenIterator;

public class Block extends Stmt {
    public Block() {
        super(ASTNodeTypes.BLOCK, "block");
    }

    public static ASTNode parse(PeekTokenIterator it) throws ParseException {
        it.nextMatch("{");
        var block = new Block();
        ASTNode stmt = null;
        while( (stmt = Stmt.parseStmt(it)) != null) {
            block.addChild(stmt);
        }
        it.nextMatch("}");
        return block;

    }




}

File: js/src/parser/ast/Block.js
const ASTNodeTypes = require('./ASTNodeTypes')
const {Stmt} = require('./index')
class Block extends Stmt {
    constructor() {
        super(ASTNodeTypes.BLOCK, 'block')
    }

    static parse(it) {
        it.nextMatch("{")
        const block = new Block()
        const stmt = null

        while( (stmt = Stmt.parseStmt(it)) != null) {
            block.addChild(stmt);
        }
        it.nextMatch("}")
        return block
    }
}


module.exports = Block 